add-on/panels/js/panel.jsx

```diff
@@ -891,6 +883,103 @@ loop.panel = (function(_, mozL10n) {
     }
   });
 
+  var SharePanelView = React.createClass({
+    mixins: [
+      Backbone.Events,
+      sharedMixins.DocumentVisibilityMixin,
+      sharedMixins.WindowCloseMixin
+    ],
+
+    propTypes: {
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired,
+      // Force render for testing purpose
+      forceRender: React.PropTypes.bool,
+      store: React.PropTypes.instanceOf(loop.store.RoomStore).isRequired
+    },
+
+    getInitialState: function() {
+      return this.props.store.getStoreState();
+    },
+
+    componentWillMount: function() {
+      loop.request("GetLoopPref", "facebook.enabled").then(result => {
+        this.setState({
+          facebookEnabled: result
+        });
+      });
+    },
+
+    componentDidMount: function() {
+      this.listenTo(this.props.store, "change", this._onStoreStateChanged);
+    },
+
+    componentWillUnmount: function() {
+      this.stopListening(this.props.store);
+    },
+
+    _onStoreStateChanged: function() {
+      this.setState(this.props.store.getStoreState());
+    },
+
+    onDocumentHidden: function() {
+      this.state.shouldOpenSharePanel && this._closeSharePanel();
+    },
+
+    componentWillUpdate: function(nextProps, nextState) {
+      // If we've just created a room, close the panel - the store will open
+      // the room.
+      if (this.state.pendingCreation &&
+          !nextState.pendingCreation && !nextState.error) {
+        this.setState({
+          shouldOpenSharePanel: true
+        });
+      }
+    },
+
+    _closeSharePanel: function() {
+      this.setState({
+        shouldOpenSharePanel: false
+      });
+
+      this.openRoom();
+    },
+
+    openRoom: function() {
+      var activeRoom = this.state.activeRoom;
+      this.props.dispatcher.dispatch(new sharedActions.OpenRoom({
+        roomToken: activeRoom.roomToken
+      }));
+
+      this.closeWindow();
+    },
+
+    render: function() {
+      var sharePanelClasses = classNames({
+        "share-panel-container": true,
+        "share-panel-open": this.state.shouldOpenSharePanel || this.props.forceRender
```

Comment:
I would prefer if we didn't render the share panel at all if we're not displaying it.